### PR TITLE
Use hex representation of crate hash in YK labels

### PR DESCRIPTION
**Pull request by ptersilie**

### CODE REVIEW

File: src/librustc_codegen_ssa/mir/block.rs [hunk 1]
```diff
@@ -147,7 +147,7 @@ impl<'a, 'tcx> TerminatorCodegenHelper<'a, 'tcx> {
                 if bx.cx().tcx().sess.opts.cg.tracer.sir_labels() && bx.cx().has_debug() {
                     let did = fx.instance.def.def_id();
                     let crate_hash = bx.cx().tcx().crate_hash(did.krate).as_u64();
-                    let lbl_name = CString::new(format!("__YK_RET_{}_{}_{}", crate_hash,
+                    let lbl_name = CString::new(format!("__YK_RET_{:x?}_{}_{}", crate_hash,
```
Comment (vext01) [inline src/librustc_codegen_ssa/mir/block.rs]: Why the `?` in the format string? `u64` should implement `Display`, right?

Comment (ptersilie) [reply src/librustc_codegen_ssa/mir/block.rs]: Whoops, that's a typo.

**Comment by ptersilie** [comment]
Fixed.

**Comment by vext01** [comment]
OK, this looks good. Does it affect anything in the `yk` repo?

**Comment by ptersilie** [comment]
There's no hardware tracing related stuff in the yk repo yet, so no.

**Comment by vext01** [comment]
Please squash.

**Comment by ptersilie** [comment]
Squashed!

**Pull request merged**